backend/app/api/v1/endpoints/devices.py

```
@@ -0,0 +1,94 @@
+from app.schemas.device import SmartDevice, SmartDeviceCreate, SmartDeviceUpdate
+from fastapi import APIRouter, HTTPException
+
+router = APIRouter()
+
+
+@router.get("/", response_model=List[SmartDevice])
+async def read_devices(
+    db: deps.SessionDep,
+    skip: int = 0,
+    limit: int = 100,
+    user_id: int = None,
```

Ruff (0.14.5)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)